### ISSUE TITLE
[prefer-inject] should not be triggered when constructor calls super()

### ISSUE THREAD

**Issue by e-oz** [angular-eslint/angular-eslint#2473]
Because "this" can not be accessed in the constructor before the super() call.

For this case:
```
abstract class X {
   constructor(protected readonly url: string) {
   }
}

class Y extends X {
   constructor(@Inject(ACCOUNT_URL_TOKEN) url: string) {
      super(url);
   }
}
```
Replacing to
```
class Y extends X {
   apiUrl = inject(ACCOUNT_URL_TOKEN);
   constructor() {
      super(this.apiUrl);
   }
}
```
gives an error:
```
TS17009: super must be called before accessing this in the constructor of a derived class.
```

**Comment by e-oz** [comment]
No, you can't use  inject() with super(), lol.
But ok, I'll just disable this rule.

**Comment by e-oz** [comment]
So you are saying that abstract classes and inheritance is discouraged, because it should not work. I get it! ;)

**Comment by JamesHenry** [comment]
Like many people, it seems that you are overly concerned with using `eslint-disable` comments in your code, but they are not some evil thing.

Having to add a comment next to this code and then document inline - "hey, I know this isn't recommended by the Angular Style Guide, but we're doing it this way for good reasons A,B,C so disabling the lint rule specifically here" is a _good_ thing. Much better than disabling the rule, because it allows for existing patterns to remain, and now be better documented, but for developers to think twice about introducing net new code which deviates from the recommendations of the Angular Team.

If you want to get into a debate with them about inheritance and DI patterns and they revise their guidance off the back of it, great, we can revisit then.

**Comment by JamesHenry** [comment]
But the default should remain to align with the style guide
